packages/stack/blockchain/test/blockchain.spec.js

```diff
@@ -0,0 +1,534 @@
+        const cb = await blockchain.plugins.mock.consoleCommand('log blockchain off');
+        sinon.assert.calledWith(cb, '123');
+        blockchain.events.assert.commandHandlerCalled('logs:ethereum:disable');
+      });
+    });
+
+    describe('getProviderFromTemplate', () => {
+      test('it should get a HTTP provider if endpoint doesn\'t start with \'ws\'', async () => {
+        blockchain.getProviderFromTemplate(endpoint);
+        sinon.assert.calledWith(blockchain.Web3, endpoint);
+      });
+      test('it should get a WS provider if endpoint starts with \'ws\'', async () => {
+        const endpoint = 'ws://';
+        blockchain.getProviderFromTemplate(endpoint);
+        sinon.assert.calledWith(blockchain.Web3.providers.WebsocketProvider, endpoint, {
+          headers: { Origin: constants.embarkResourceOrigin }
+        });
+      });
+    });
+
+    describe('addArtifactFile', () => {
+      test('it should return with nothing if not enabled', () => {
+        blockchain.blockchainConfig.enabled = false;
+        const cb = sinon.fake();
+        const contractsConfigFn = sinon.fake.yields(null, {});
+        blockchain.events.setCommandHandler('config:contractsConfig', contractsConfigFn);
+
+        blockchain.addArtifactFile(null, cb);
+        assert(cb.called);
+        blockchain.events.assert.commandHandlerNotCalled('config:contractsConfig');
+      });
+      test('it should replace $EMBARK with proxy endpoint', async () => {
+        const cb = sinon.fake();
+        const endpoint = "endpoint2";
+        const contractsConfig = {
+          dappConnection: ['endpoint1', '$EMBARK', 'endpoint3'],
+          dappAutoEnable: true
+        };
+        const contractsConfigFn = sinon.fake.yields(null, contractsConfig);
+        const networkId = '1337';
+        const config = {
+          provider: 'web3',
+          dappConnection: ['endpoint1', 'endpoint2', 'endpoint3'],
+          library: 'embarkjs',
+          dappAutoEnable: contractsConfig.dappAutoEnable,
+          warnIfMetamask: blockchain.blockchainConfig.isDev,
+          blockchainClient: blockchain.blockchainConfig.client,
+          networkId
+        };
+        const params = {
+          path: [blockchain.embarkConfig.generationDir, 'config'],
+          file: 'blockchain.json',
+          format: 'json',
+          content: config
+        };
+        const pipelineRegisterFn = sinon.fake.yields(params, cb);
+        const networkIdFn = sinon.fake.yields(null, networkId);
+        blockchain.events.setCommandHandler('config:contractsConfig', contractsConfigFn);
+        blockchain.events.setCommandHandler('blockchain:networkId', networkIdFn);
+        blockchain.events.setCommandHandler('proxy:endpoint:get', sinon.fake.yields(null, endpoint));
+        blockchain.events.setCommandHandler('pipeline:register', pipelineRegisterFn);
+
+        await blockchain.addArtifactFile(null, cb);
```

Comment:
After rebase `addArtifactFile()` is now `async`. This is because we're now requesting the `networkId` to put it in the artifacts. As a result, we need to `await` it here, otherwise the tests will fail as assertions run too early.